src/index.ts

```diff
@@ -172,4 +225,88 @@ export class I18N {
         }
         return pluralizer || pluralizerEn;
     }
+
+    private getTranslationData(args: {
+        keysetName: string;
+        key: string;
+        lang?: string;
+        params?: Params;
+    }): TranslationData {
+        const {lang, key, keysetName, params} = args;
+        const languageData = this.getLanguageData(lang);
+
+        if (typeof languageData === 'undefined') {
+            return {details: {code: ErrorCode.NoLanguageData}};
+        }
+
+        if (Object.keys(languageData).length === 0) {
+            return {
+                fallbackText: key,
+                details: {code: ErrorCode.EmptyLanguageData},
+            };
+        }
+
+        const keyset = languageData[keysetName];
+
+        if (!keyset) {
+            return {
+                fallbackText: key,
+                details: {code: ErrorCode.KeysetNotFound, keysetName},
+            };
+        }
+
+        if (Object.keys(keyset).length === 0) {
+            return {
+                fallbackText: key,
+                details: {code: ErrorCode.EmptyKeyset, keysetName},
+            };
+        }
+
+        const keyValue = keyset && keyset[key];
+        const result: TranslationData = {};
+
+        if (typeof keyValue === 'undefined') {
+            return {
+                fallbackText: key,
+                details: {code: ErrorCode.MissingKey, keysetName, key},
+            };
+        }
+
+        if (Array.isArray(keyValue)) {
+            if (keyValue.length < 3) {
+                return {
+                    fallbackText: key,
+                    details: {code: ErrorCode.MissingKeyPlurals, keysetName, key},
+                };
+            }
+
+            const count = Number(params?.count);
+
+            if (Number.isNaN(count)) {
+                return {
+                    fallbackText: key,
+                    details: {code: ErrorCode.MissingKeyParamsCount, keysetName, key},
+                };
+            }
+
+            const pluralizer = this.getLanguagePluralizer(lang);
+            result.text = keyValue[pluralizer(count, PluralForm)] || keyValue[PluralForm.Many];
+
```

Comment:
```suggestion
            if (result.text === undefined) {
                return {
                    fallbackText: key,
                    details: {code: ErrorCode.MissingKeyPlurals, keysetName, key},
                };
            }

```

let's add `"noUncheckedIndexedAccess": true` to tsconfig